src/plugins/intel_gpu/src/runtime/ocl/ocl_stream.cpp

```diff
@@ -304,6 +304,16 @@ event::ptr ocl_stream::enqueue_kernel(kernel& kernel,
     try {
         _command_queue.enqueueNDRangeKernel(kern, cl::NullRange, global, local, dep_events_ptr, set_output_event ? &ret_ev : nullptr);
     } catch (cl::Error const& err) {
+        /// WA: Force exit. Any opencl api call can be hang after CL_OUT_OF_RESOURCES.
+        if (err.err() == CL_OUT_OF_RESOURCES) {
+            std::cerr << "[GPU Plugin] force exit.\n"
```

Comment:
1. `[GPU] ... `
2. Does it impact dGPU only? If yes, I think we need to check device type here and just rethrow exception on igpu.

Reply:
It's dGPU only. In iGPU case, the app will be killed with 'Out of memory' by host kernel side before gpu plugin gets 'CL_OUT_OF_RESOURCES'.